web/app/src/utils/strings.ts

```diff
@@ -0,0 +1,7 @@
+export function ltrim(str: string, charlist: string) {
+  return str.replace(new RegExp('^[' + charlist + ']+'), '');
+}
+
+export function rtrim(str: string, charlist: string) {
+  return str.replace(new RegExp('[' + charlist + ']+$'), '');
+}
```

Comment:
what about a kind of url.join(...) (like path.join but for url) method instead ?

Reply:
Sorry, I don't get what you mean specifically here.

Reply:
it looks like ltrim/rtrim functions only deserves the api url generator:
```
export function apiUri(path: string): string {
  return [
    rtrim(AppConfig.API_BASE_URI, '/'),
    ltrim(path, '/'),
  ].join('/');
}
```
i wonder if a url joiner (just like https://nodejs.org/api/path.html#pathjoinpaths) could more elegantly deal with it: ```
export function apiUri(path: string): string {
  return url.join(AppConfig.API_BASE_URI, path)
}
```